File: public_site/tools.py
"""Public operator tools served from noetfeld-os."""

from __future__ import annotations

from dataclasses import dataclass

PUBLIC_ORIGIN = "https://api.noetfield.com"


@dataclass(frozen=True)
class ToolPage:
    slug: str
    title: str
    description: str
    kicker: str
    hero_headline: str
    hero_subheadline: str
    form_html: str = ""
    presets: tuple[tuple[str, str], ...] = ()
    math_line: str = ""
    notes: tuple[tuple[str, str], ...] = ()
    kind: str = "calc"
    hide_amount: bool = False


HUB = ToolPage(
    slug="hub",
    title="Free operator tools // Noetfield OS",
    description="One-minute checks that will tell you to leave a process alone. No signup. Nothing stored.",
    kicker="Free · no signup · nothing stored",
    hero_headline="Most operators can name the process that annoys them. Almost none can name what it costs.",
    hero_subheadline="Five one-minute checks. Conservative math. An honest leave-it-alone line. If the number is a hobby, the page says so instead of selling you a fix.",
    kind="hub",
)

PAGES: dict[str, ToolPage] = {
    "quiet-leak": ToolPage(
        slug="quiet-leak",
        title="Quiet leak // What is this process costing you?",
        description="Four inputs. Rate times 1.3 times 48 weeks. Under $3,000 a year, leave it alone.",
        kicker="Same math as the operator post",
        hero_headline="What is this process costing you?",
        hero_subheadline="Pick the quiet one. Double entry. Status copied into a tracker. A draft rewritten after the model said it was done.",
        presets=(
            ('{"touches":10,"minutes":8,"rate":45,"people":3}', "Double data entry"),
            ('{"touches":6,"minutes":15,"rate":40,"people":2}', "Invoice chase"),
            ('{"touches":12,"minutes":5,"rate":38,"people":4}', "CRM copy-paste"),
        ),
        form_html="""
       <label>Times a week someone touches it
        <input type="number" name="touches" min="0" step="0.5" value="8" inputmode="decimal" required />
       </label>
       <label>Minutes per touch
        <span class="hint">Include the context switch.</span>
        <input type="number" name="minutes" min="0" step="0.5" value="12" inputmode="decimal" required />
       </label>
       <label>Hourly rate before overhead (CAD)
        <input type="number" name="rate" min="0" step="1" value="45" inputmode="decimal" required />
       </label>
       <label>How many people do this
        <input type="number" name="people" min="0" step="1" value="3" inputmode="decimal" required />
       </label>
        """,
        math_line="Annual cost = touches × (minutes ÷ 60) × rate × 1.3 × people × 48. Under $3,000, leave it alone.",
        notes=(
            (
                "People undercount touches by about half",
                "Count one real day before you estimate. The number you type from memory is usually the complaint, not the work.",
            ),
            (
                "The expensive leaks are the quiet ones",
                "Nobody files a ticket for the copy between two systems. Everyone accepted it years ago. That is why it survives.",
            ),
        ),
    ),
    "ai-spend": ToolPage(
        slug="ai-spend",
        title="AI spend // Invoice versus workflow",
        description="What share of the AI invoice maps to a named workflow and a named accepter.",
        kicker="Invoice versus workflow",
        hero_headline="Most teams can name the AI invoice. Almost none can name which workflow created it.",
        hero_subheadline="If spend is small and one team owns it, a spreadsheet is enough. If several teams are in it and you cannot attribute 20% of the bill, the leak is explanation, not tokens.",
        presets=(
            ('{"monthly":800,"attributed":80,"teams":1,"named":"yes"}', "One team, small bill"),
            ('{"monthly":12000,"attributed":10,"teams":6,"named":"no"}', "Several teams, unexplained"),
        ),
        form_html="""
       <label>Monthly AI / Copilot spend (CAD)
        <input type="number" name="monthly" min="0" step="50" value="4000" inputmode="decimal" required />
       </label>
       <label>Share you can attribute to a named workflow (%)
        <input type="number" name="attributed" min="0" max="100" step="1" value="15" inputmode="decimal" required />
       </label>
       <label>Teams using it
        <input type="number" name="teams" min="1" step="1" value="3" inputmode="decimal" required />
       </label>
       <label>Does a named person accept the output before it leaves?
        <select name="named" required>
         <option value="no" selected>No</option>
         <option value="yes">Yes</option>
        </select>
       </label>
        """,
        math_line="Unattributed annual = monthly × (1 − attributed share) × 12. Leave it alone under $1,500 a month with one team.",
        notes=(
            (
                "Licensed Copilot is not the whole bill",
                "People count seats and forget personal ChatGPT. That unofficial line is often larger, and it never shows up in the invoice meeting.",
            ),
            (
                "The draft that shipped with no name on it",
                "Nobody files a ticket for that. So it survives. That is the expensive leak.",
            ),
        ),
    ),
    "who-accepted": ToolPage(
        slug="who-accepted",
        title="Who accepted // Chat log versus process",
        description="If you cannot name who accepted the last AI output, you have a chat log, not a process.",
        kicker="Chat log versus process",
        hero_headline="If you cannot name who accepted the last AI output, you do not have a process.",
        hero_subheadline="A process has a named person, a pass/fail check, and a reason you can open later. The page will tell you to stop shopping if you already have that.",
        presets=(
            ('{"deliverables":8,"signed":95,"minutes":5,"rate":55,"replay":"yes"}', "Already signed"),
            ('{"deliverables":30,"signed":10,"minutes":25,"rate":65,"replay":"no"}', "Chat drafts, heavy rewrite"),
        ),
        form_html="""
       <label>AI-assisted deliverables per week
        <input type="number" name="deliverables" min="0" step="1" value="20" inputmode="decimal" required />
       </label>
       <label>Share a named person signs (%)
        <input type="number" name="signed" min="0" max="100" step="1" value="25" inputmode="decimal" required />
       </label>
       <label>Minutes of redo on an unsigned item
        <input type="number" name="minutes" min="0" step="1" value="18" inputmode="decimal" required />
       </label>
       <label>Hourly rate before overhead (CAD)
        <input type="number" name="rate" min="0" step="1" value="55" inputmode="decimal" required />
       </label>
       <label>Can you replay why the last one passed?
        <select name="replay" required>
         <option value="no" selected>No</option>
         <option value="yes">Yes</option>
        </select>
       </label>
        """,
        math_line="Unsigned volume × redo minutes × loaded rate × 48 weeks. Leave it alone at 90% signed with a replayable why.",
        notes=(
            (
                "The builder must not grade itself",
                "If the same person or model that produced the draft also marks it done, you do not have a check. You have a hope.",
            ),
            (
                "Redo is the bill you already pay",
                "Minutes spent rewriting finished drafts is the real cost. The tool prices that, not a fantasy saving.",
            ),
        ),
    ),
    "copilot-seats": ToolPage(
        slug="copilot-seats",
        title="Copilot seats // Two numbers, not one",
        description="Unused licenses and ungoverned use, counted separately on purpose.",
        kicker="Two numbers, not one",
        hero_headline="You are paying for seats. You are not paying for a decision trail.",
        hero_subheadline="Unused licenses are an adoption problem. Ungoverned use is an explanation problem. Showing only one of those numbers is how a page stays dishonest.",
        presets=(
            ('{"licensed":12,"used":11,"hours":2,"rate":45,"seat":360}', "Small team, used"),
            ('{"licensed":200,"used":40,"hours":6,"rate":70,"seat":360}', "Wide rollout, thin use"),
        ),
        form_html="""
       <label>Licensed seats
        <input type="number" name="licensed" min="0" step="1" value="80" inputmode="decimal" required />
       </label>
       <label>Seats used last week
        <input type="number" name="used" min="0" step="1" value="35" inputmode="decimal" required />
       </label>
       <label>Hours per used seat per week
        <input type="number" name="hours" min="0" step="0.5" value="4" inputmode="decimal" required />
       </label>
       <label>Hourly rate before overhead (CAD)
        <input type="number" name="rate" min="0" step="1" value="55" inputmode="decimal" required />
       </label>
       <label>Annual cost per seat (CAD)
        <span class="hint">Default 360 for Copilot-class licensing. Change it if you know the real number.</span>
        <input type="number" name="seat" min="0" step="10" value="360" inputmode="decimal" required />
       </label>
        """,
        math_line="Unused waste = unused seats × annual seat cost. Ungoverned use = used seats × hours × loaded rate × 48 weeks.",
        notes=(
            (
                "Unused seats under about 10",
                "That is adoption. Do not buy a control plane to fix a habit. Ask why people are not in the tool.",
            ),
            (
                "Used seats with no named accepter",
                "That is the expensive line. Hours are being spent and nobody can replay why an output left the building.",
            ),
        ),
    ),
    "board-five": ToolPage(
        slug="board-five",
        title="Board five // Yes or no only",
        description="Five yes or no questions. Score 0 or 1: do not buy. The tool will say that out loud.",
        kicker="Yes or no only",
        hero_headline="The board questions you either can answer or you cannot.",
        hero_subheadline="No scores dressed up as science. Check what you can actually name today. The page will tell you not to buy.",
        presets=(("{}", "Clear all"),),
        form_html="""
       <div class="tools-checks">
        <label><input type="checkbox" name="workflow" /> Can you name the workflow?</label>
        <label><input type="checkbox" name="owner" /> Can you name the owner?</label>
        <label><input type="checkbox" name="spend" /> Can you name last month’s spend for it?</label>
        <label><input type="checkbox" name="failed" /> Can you name the last time it failed?</label>
        <label><input type="checkbox" name="accepted" /> Can you name who accepted the last output?</label>
       </div>
        """,
        math_line="Score is a count of yes answers. 0 to 1 leave it. 2 to 3 Copilot Readiness if procurement needs a file. 4 to 5 Trust Brief only if you need the memo.",
        notes=(
            (
                "0 or 1: do not buy",
                "Name one workflow and who accepts its output. Then come back. A diagnostic sold into a blank page is theatre.",
            ),
            (
                "4 or 5: a memo can be defended",
                "Trust Brief is a six-week policy map, not a product tour. Use it when the board needs that memo, not before.",
            ),
        ),
        hide_amount=True,
    ),
    "embed": ToolPage(
        slug="embed",
        title="Embed these tools // Advisors",
        description="One iframe. No signup. We do not set cookies on your visitors and we do not store their numbers.",
        kicker="Advisors · free",
        hero_headline="Put the check on your own page.",
        hero_subheadline="One iframe. No signup. We do not set cookies on your visitors and we do not store their numbers. Keep the Noetfield link under the frame so people can open the full page.",
        kind="embed",
    ),
}

CARDS: tuple[tuple[str, str, str, str], ...] = (
    ("quiet-leak", "01", "Quiet leak", "Price one manual process. Honest hobby line at $3,000 a year."),
    ("ai-spend", "02", "AI spend you cannot explain", "What share of the invoice maps to a named workflow."),
    ("who-accepted", "03", "Who accepted this", "Chat log versus a named person and a replayable why."),
    ("copilot-seats", "04", "Copilot seats", "Unused licenses and ungoverned use, shown as two numbers."),
    ("board-five", "05", "Five board questions", "Yes or no. The tool will tell you not to buy."),
    ("embed", "06", "Embed for advisors", "One iframe. No tracking of your visitors."),
)

EMBED_BLOCKS: tuple[tuple[str, str], ...] = (
    ("quiet-leak", "Quiet leak"),
    ("ai-spend", "AI spend"),
    ("who-accepted", "Who accepted"),
    ("copilot-seats", "Copilot seats"),
    ("board-five", "Five board questions"),
)


def canonical(slug: str) -> str:
    if slug in ("", "hub"):
        return f"{PUBLIC_ORIGIN}/tools/"
    return f"{PUBLIC_ORIGIN}/tools/{slug}/"


def get_page(slug: str | None) -> ToolPage | None:
    if slug in (None, "", "hub"):
        return HUB
    return PAGES.get(slug or "")


__all__ = [
    "CARDS",
    "EMBED_BLOCKS",
    "HUB",
    "PAGES",
    "PUBLIC_ORIGIN",
    "ToolPage",
    "canonical",
    "get_page",
]

File: public_site/static/tools.js
(function () {
  var LOAD = 1.3;
  var WEEKS = 48;
  var HOBBY = 3000;
  var NOTICE = "Nothing is posted. Nothing is stored.";

  function money(n) {
    var v = Number.isFinite(n) && n > 0 ? n : 0;
    return new Intl.NumberFormat("en-CA", {
      style: "currency",
      currency: "CAD",
      maximumFractionDigits: 0,
    }).format(Math.round(v));
  }

  function num(form, name) {
    return Number(form.elements[name] && form.elements[name].value ? form.elements[name].value : 0);
  }

  function sel(form, name) {
    var el = form.elements[name];
    return el ? String(el.value || "") : "";
  }

  function checked(form, name) {
    var el = form.elements[name];
    return !!(el && el.checked);
  }

  function processCost(touches, minutes, rate, people) {
    return Number(touches) * (Number(minutes) / 60) * Number(rate) * LOAD * Number(people) * WEEKS;
  }

  var CTAS = {
    app: { href: "https://app.noetfield.com/", label: "Open the app" },
    readiness: { href: "https://www.noetfield.com/copilot/pilot/", label: "Copilot pilot" },
    brief: { href: "https://www.noetfield.com/trust-brief/", label: "Trust Brief" },
    intake: { href: "https://www.noetfield.com/trust-brief/intake/", label: "Trust Brief intake" },
    calculator: { href: "https://sourcea.app/calculator", label: "Full process-cost calculator" },
    tools: { href: "/tools/", label: "See the other checks" },
  };

  function compute(tool, form) {
    if (tool === "quiet-leak") {
      var cost = processCost(num(form, "touches"), num(form, "minutes"), num(form, "rate"), num(form, "people"));
      if (cost < HOBBY) {
        return {
          kind: "leave",
          amount: money(cost) + " / year",
          headline: "Leave it alone.",
          body: "Under $3,000 a year, automating this is a hobby, not an investment. The page says that out loud instead of selling you a fix.",
          extra: "Count one real day before you trust the estimate. People undercount touches by about half.",
          cta: "calculator",
        };
      }
      if (cost < 15000) {
        return {
          kind: "look",
          amount: money(cost) + " / year",
          headline: "Real money. Still not a platform.",
          body: "Price one leak first. If the number is real, start with one named goal and a check that can fail.",
          extra: "",
          cta: "app",
        };
      }
      return {
        kind: "act",
        amount: money(cost) + " / year",
        headline: "This leak is large enough to name.",
        body: "Give this process as a goal in the open alpha. Plan, run, check, stop for your decision. Founder-operated. No fake customer logos.",
        extra: "",
        cta: "app",
      };
    }

    if (tool === "ai-spend") {
      var monthly = Math.max(0, num(form, "monthly"));
      var share = Math.min(100, Math.max(0, num(form, "attributed"))) / 100;
      var teams = Math.max(1, num(form, "teams") || 1);
      var named = sel(form, "named") === "yes";
      var annual = monthly * (1 - share) * 12;
      var amount = money(annual) + " / year unattributed";
      if (monthly < 1500 && teams <= 1) {
        return {
          kind: "leave",
          amount: amount,
          headline: "Leave it alone.",
          body: "Under about $1,500 a month and one team, a spreadsheet is enough. Do not buy a governance stack for a hobby.",
          extra: "",
          cta: "tools",
        };
      }
      if (named && share >= 0.2) {
        return {
          kind: "leave",
          amount: amount,
          headline: "You already have the bones.",
          body: "Named workflow and named accepter. Do not buy a platform tour. Tighten the receipt, then stop shopping.",
          extra: "",
          cta: "app",
        };
      }
      if (share < 0.2) {
        return {
          kind: "act",
          amount: amount,
          headline: "The leak is explanation, not tokens.",
          body: "Under 20% attributed, nobody can defend the bill. Name one workflow, one owner, and who accepts output before it leaves.",
          extra: "People count licensed Copilot and forget personal ChatGPT. That unofficial line is often larger.",
          cta: "readiness",
        };
      }
      return {
        kind: "look",
        amount: amount,
        headline: "Attribute more before you buy more.",
        body: "Spend is large enough to care. The next honest step is a named workflow and a named accepter, not another seat.",
        extra: "",
        cta: "app",
      };
    }

    if (tool === "who-accepted") {
      var deliverables = Math.max(0, num(form, "deliverables"));
      var signed = Math.min(100, Math.max(0, num(form, "signed"))) / 100;
      var minutes = Math.max(0, num(form, "minutes"));
      var rate = Math.max(0, num(form, "rate"));
      var replay = sel(form, "replay") === "yes";
      var redo = deliverables * (1 - signed) * (minutes / 60) * rate * LOAD * WEEKS;
      var amount = money(redo) + " / year of unsigned redo";
      if (signed >= 0.9 && replay) {
        return {
          kind: "leave",
          amount: amount,
          headline: "Leave it alone.",
          body: "A named signer and a replayable why. You do not need another copilot. Keep the receipt. Stop shopping.",
          extra: "",
          cta: "tools",
        };
      }
      if (signed < 0.5 || !replay) {
        return {
          kind: "act",
          amount: amount,
          headline: "That is a chat log, not a process.",
          body: "Name who accepts, keep the reason, and let a check fail. One real goal in the open alpha is enough to try that.",
          extra: "The builder must not grade itself. If the model that wrote the draft also marks it done, you have a hope, not a check.",
          cta: "app",
        };
      }
      return {
        kind: "look",
        amount: amount,
        headline: "Close the replay gap.",
        body: "Signing without a why still fails a board question. Record the pass/fail reason before you buy more seats.",
        extra: "",
        cta: "app",
      };
    }

    if (tool === "copilot-seats") {
      var licensed = Math.max(0, num(form, "licensed"));
      var used = Math.min(licensed, Math.max(0, num(form, "used")));
      var hours = Math.max(0, num(form, "hours"));
      var wage = Math.max(0, num(form, "rate"));
      var seat = Math.max(0, num(form, "seat"));
      var unused = Math.max(0, licensed - used);
      var waste = unused * seat;
      var ungoverned = used * hours * wage * LOAD * WEEKS;
      var amount = money(waste) + " unused licenses · " + money(ungoverned) + " ungoverned use / year";
      if (unused < 10 && licensed <= 20) {
        return {
          kind: "leave",
          amount: amount,
          headline: "Fix adoption. Do not buy policy.",
          body: "Unused seats under about 10 is not a governance purchase. Turn unused licenses off, or train the people who have them.",
          extra: "Showing only the unused-license number is how a post stays dishonest. The used-seat line is usually larger.",
          cta: "tools",
        };
      }
      if (used >= 10 && hours >= 2) {
        return {
          kind: "act",
          amount: amount,
          headline: "You are paying for labor with no trail.",
          body: "The used-seat line is the one that matters. Name the workflow, the owner, and who accepts output.",
          extra: "",
          cta: "brief",
        };
      }
      return {
        kind: "look",
        amount: amount,
        headline: "Show both numbers to finance.",
        body: "License waste is visible. Ungoverned use is usually larger and quieter. Do not let a seat-optimization slide hide the second number.",
        extra: "",
        cta: "readiness",
      };
    }

    if (tool === "board-five") {
      var n =
        (checked(form, "workflow") ? 1 : 0) +
        (checked(form, "owner") ? 1 : 0) +
        (checked(form, "spend") ? 1 : 0) +
        (checked(form, "failed") ? 1 : 0) +
        (checked(form, "accepted") ? 1 : 0);
      if (n <= 1) {
        return {
          kind: "leave",
          amount: n + " / 5",
          headline: "Do not buy.",
          body: "You are not ready for a diagnostic. Name the workflow and the owner first. Open the app if you want one real goal with a check that can fail.",
          extra: "",
          cta: "app",
        };
      }
      if (n <= 3) {
        return {
          kind: "look",
          amount: n + " / 5",
          headline: "Procurement may need a file. You do not need a tour.",
          body: "Copilot Readiness is the pack that can be filed. The missing yeses are still the work: spend, last failure, or who accepted.",
          extra: "",
          cta: "readiness",
        };
      }
      return {
        kind: "act",
        amount: n + " / 5",
        headline: "Trust Brief only if you need a board memo.",
        body: "You can already answer the room. Buy a memo if the board needs paper. Do not buy another copilot to feel busy.",
        extra: "",
        cta: "intake",
      };
    }

    return null;
  }

  function applyPreset(form, raw) {
    var data;
    try {
      data = JSON.parse(raw);
    } catch {
      return;
    }
    Object.keys(data).forEach(function (key) {
      var el = form.elements[key];
      if (!el) return;
      if (el.type === "checkbox") el.checked = !!data[key];
      else el.value = data[key];
    });
    if (Object.keys(data).length === 0) {
      Array.prototype.forEach.call(form.elements, function (el) {
        if (el.type === "checkbox") el.checked = false;
      });
    }
  }

  function applyParams(form) {
    var params = new URLSearchParams(window.location.search);
    Array.prototype.forEach.call(form.elements, function (el) {
      if (!el.name || !params.has(el.name)) return;
      if (el.type === "checkbox") el.checked = params.get(el.name) === "1" || params.get(el.name) === "true";
      else el.value = params.get(el.name);
    });
  }

  function shareUrl(form) {
    var url = new URL(window.location.href);
    url.searchParams.delete("embed");
    Array.prototype.forEach.call(form.elements, function (el) {
      if (!el.name) return;
      if (el.type === "checkbox") url.searchParams.set(el.name, el.checked ? "1" : "0");
      else url.searchParams.set(el.name, String(el.value || ""));
    });
    return url.toString();
  }

  function flash(btn, text) {
    if (!btn) return;
    var prior = btn.textContent;
    btn.textContent = text;
    setTimeout(function () {
      btn.textContent = prior;
    }, 1600);
  }

  function render(tool, form) {
    var result = compute(tool, form);
    if (!result) return;
    var box = document.getElementById("nf-tools-result");
    var amount = document.querySelector("[data-result-amount]");
    var headline = document.querySelector("[data-result-headline]");
    var body = document.querySelector("[data-result-body]");
    var extra = document.querySelector("[data-result-extra]");
    var cta = document.querySelector("[data-result-cta]");
    if (box) box.setAttribute("data-kind", result.kind);
    if (amount && !amount.hidden) amount.textContent = result.amount;
    if (amount && amount.hidden && tool === "board-five") {
      amount.hidden = false;
      amount.textContent = result.amount;
    }
    if (headline) headline.textContent = result.headline;
    if (body) body.textContent = result.body;
    if (extra) {
      extra.textContent = result.extra || "";
      extra.hidden = !result.extra;
    }
    var dest = CTAS[result.cta] || CTAS.tools;
    if (cta) {
      cta.setAttribute("href", dest.href);
      cta.textContent = dest.label;
    }
  }

  function bootEmbedBlocks() {
    document.querySelectorAll("[data-embed-src]").forEach(function (block) {
      var src = block.getAttribute("data-embed-src");
      var code = '<iframe src="' + src + '" title="Noetfield tool" width="100%" height="740" style="border:0;border-radius:12px" loading="lazy"></iframe>';
      var slot = block.querySelector("code");
      if (slot) slot.textContent = code;
      var btn = block.querySelector("[data-copy-embed]");
      if (btn) {
        btn.addEventListener("click", function () {
          navigator.clipboard.writeText(code).then(function () {
            flash(btn, "Copied");
          }).catch(function () {
            window.prompt("Copy this embed", code);
          });
        });
      }
    });
  }

  function boot() {
    var params = new URLSearchParams(window.location.search);
    if (params.get("embed") === "1") document.body.classList.add("nf-tools--embed");
    bootEmbedBlocks();
    var tool = document.body.getAttribute("data-tool") || "";
    var form = document.getElementById("nf-tools-form");
    if (!form) return;
    applyParams(form);
    render(tool, form);
    form.addEventListener("input", function () {
      render(tool, form);
    });
    form.addEventListener("change", function () {
      render(tool, form);
    });
    form.addEventListener("submit", function (event) {
      event.preventDefault();
    });
    document.querySelectorAll("[data-preset]").forEach(function (btn) {
      btn.addEventListener("click", function () {
        applyPreset(form, btn.getAttribute("data-preset") || "{}");
        render(tool, form);
      });
    });
    var share = document.getElementById("nf-tools-share");
    if (share) {
      share.addEventListener("click", function () {
        var href = shareUrl(form);
        navigator.clipboard.writeText(href).then(function () {
          flash(share, "Link copied");
        }).catch(function () {
          window.prompt("Copy this link", href);
        });
      });
    }
    var copy = document.getElementById("nf-tools-copy");
    if (copy) {
      copy.addEventListener("click", function () {
        var headline = document.querySelector("[data-result-headline]");
        var body = document.querySelector("[data-result-body]");
        var amount = document.querySelector("[data-result-amount]");
        var text = [amount && amount.textContent, headline && headline.textContent, body && body.textContent, NOTICE, window.location.href]
          .filter(Boolean)
          .join("\n");
        navigator.clipboard.writeText(text).then(function () {
          flash(copy, "Copied");
        });
      });
    }
  }

  window.NF_TOOLS_MATH = { LOAD: LOAD, WEEKS: WEEKS, money: money, processCost: processCost, compute: compute };
  if (document.readyState === "loading") document.addEventListener("DOMContentLoaded", boot);
  else boot();
})();
